src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,248 @@
+    }
+
+    @Test
+    void testLongLogin_Ok() {
+        user.setLogin("username1234567890");
+        user.setPassword("password");
+        user.setAge(20);
+        assertDoesNotThrow(() -> registrationService.register(user),
+                "Login should have at least 6 characters!");
+        assertTrue(Storage.getPeople().contains(user));
+    }
+
+    @Test
+    void testZeroAge_NotOk() {
+        user.setLogin("username");
+        user.setPassword("password");
+        user.setAge(0);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user),
+                "You must be over 18 to register");
+        assertFalse(Storage.getPeople().contains(user));
+    }
+
+    @Test
+    void testAgeLessThanEighteen_NotOk() {
+        user.setLogin("username");
+        user.setPassword("password");
+        user.setAge(10);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user),
+                "You must be over 18 to register");
+        assertFalse(Storage.getPeople().contains(user));
+    }
+
+    @Test
+    void testEdgeCaseAge_NotOk() {
+        user.setLogin("username");
+        user.setPassword("password");
+        user.setAge(17);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user),
+                "You must be over 18 to register");
+        assertFalse(Storage.getPeople().contains(user));
+    }
+
+    @Test
+    void testValidAgeEdgeCase_Ok() {
+        user.setLogin("username");
+        user.setPassword("password");
+        user.setAge(18);
+        assertDoesNotThrow(() -> registrationService.register(user),
+                "You must be over 18 to register");
+        assertTrue(Storage.getPeople().contains(user));
+    }
+
+    @Test
+    void testValidAge_Ok() {
+        user.setLogin("username");
+        user.setPassword("password");
+        user.setAge(20);
+        assertDoesNotThrow(() -> registrationService.register(user),
+                "You must be over 18 to register");
+        assertTrue(Storage.getPeople().contains(user));
+    }
+
+    @Test
+    void testLongAge_Ok() {
+        user.setLogin("username");
+        user.setPassword("password");
+        user.setAge(80);
+        assertDoesNotThrow(() -> registrationService.register(user),
+                "You must be over 18 to register");
+        assertTrue(Storage.getPeople().contains(user));
+    }
+
+    @Test
+    void userAlreadyExists_NotOk() {
+        User user1 = new User();
+        user1.setLogin("khtos'");
+        user1.setAge(18);
+        user1.setPassword("yakyys");
+        User user2 = new User();
+        user2.setLogin("khtos'");
+        user2.setAge(19);
+        user2.setPassword("909090");
+        registrationService.register(user1);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user2),
+                "User already exists!");
+    }
+
+    @Test
+    void newUser_Ok() {
```

Comment:
need to fix all test names according to:
For this task use such convention: `<methodUnderTest>_<state>_<expectedBehavior>`.